src/main/java/com/thealgorithms/dynamicprogramming/SumOfSubset.java

```diff
@@ -1,18 +1,37 @@
 package com.thealgorithms.dynamicprogramming;
 
-public class SumOfSubset {
+public final class SumOfSubset {
 
-    public static boolean subsetSum(int[] arr, int num, int Key) {
-        if (Key == 0) {
+    private SumOfSubset() {
+    }
+
+    public static boolean subsetSum(int[] arr, int n, int key) {
+        if (key == 0) {
             return true;
         }
-        if (num < 0 || Key < 0) {
+        if (n < 0 || key < 0) {
             return false;
         }
+        boolean[] prev = new boolean[key + 1];
+        prev[0] = true;
+
+        if (arr[0] <= key) {
+            prev[arr[0]] = true;
+        }
 
-        boolean include = subsetSum(arr, num - 1, Key - arr[num]);
-        boolean exclude = subsetSum(arr, num - 1, Key);
+        for (int ind = 1; ind <= n; ind++) {
+            boolean[] cur = new boolean[key + 1];
+            cur[0] = true;
+
+            for (int target = 1; target <= key; target++) {
+                final boolean notTaken = prev[target];
+                final boolean taken = (arr[ind] <= target) ? prev[target - arr[ind]] : false;
+                cur[target] = notTaken || taken;
+            }
+
+            prev = cur;
+        }
```

Comment:
```suggestion
        Arrays.stream(arr).skip(1).forEach(element -> {
            boolean[] cur = new boolean[key + 1];
            cur[0] = true;

            for (int target = 1; target <= key; target++) {
                final boolean notTaken = prev[target];
                final boolean taken = (element <= target) ? prev[target - element] : false;
                cur[target] = notTaken || taken;
            }

            System.arraycopy(cur, 0, prev, 0, cur.length);
        });
```

Please make sure to `import java.util.Arrays;`.